exchanges/okx/okx_websocket.go

```diff
@@ -706,36 +901,115 @@ func (ok *Okx) wsProcessIndexCandles(respRaw []byte) error {
 	candleInterval := strings.TrimPrefix(response.Argument.Channel, candle)
 	for i := range response.Data {
 		candlesData := response.Data[i]
-		timestamp, err := strconv.ParseInt(candlesData[0], 10, 64)
-		if err != nil {
-			return err
-		}
 		myCandle := stream.KlineData{
-			Pair:      pair,
-			Exchange:  ok.Name,
-			Timestamp: time.UnixMilli(timestamp),
-			Interval:  candleInterval,
+			Pair:       pair,
+			Exchange:   ok.Name,
+			Timestamp:  time.UnixMilli(candlesData[0].Int64()),
+			Interval:   candleInterval,
+			OpenPrice:  candlesData[1].Float64(),
+			HighPrice:  candlesData[2].Float64(),
+			LowPrice:   candlesData[3].Float64(),
+			ClosePrice: candlesData[4].Float64(),
 		}
-		myCandle.OpenPrice, err = strconv.ParseFloat(candlesData[1], 64)
-		if err != nil {
-			return err
+		for i := range assets {
+			myCandle.AssetType = assets[i]
+			ok.Websocket.DataHandler <- myCandle
 		}
-		myCandle.HighPrice, err = strconv.ParseFloat(candlesData[2], 64)
-		if err != nil {
-			return err
+	}
+	return nil
+}
+
+// wsProcessPublicSpreadTicker process spread order ticker push data.
+func (ok *Okx) wsProcessPublicSpreadTicker(respRaw []byte) error {
+	var resp WsSpreadPushData
+	data := []WsSpreadPublicTicker{}
+	resp.Data = data
```

Comment:
Just a suggestion, makes sure it doesn't disappear, unless its not needed, and the rest: 
```suggestion
	resp.Data = &data
```